src/basics.ml

```diff
@@ -34,6 +35,17 @@ let to_tvar : term -> tvar = fun t ->
     “marshaled” (e.g., by the {!module:Sign} module), as this would break the
     freshness invariant of new variables. *)
 
+(** [ensure_tref t] transforms term [t] to a term with reference if it is not
+    already. *)
+let ensure_tref : term -> term = function
+  | TRef(_) as t -> t
+  | t            -> TRef(ref (Some t))
+
+(** [to_tref t] returns [r] if [t] is of the form [t = TRef(r)] and fails
+    otherwise. *)
+let to_tref : term -> term option ref = function
```

Comment:
Rename into tref_val?

Reply:
Isn't it more uniform to name the function `term -> term option ref`: `to_tref` since the function `term -> term var` is named `to_tvar`?

Reply:
to_tvar is not a well chosen name. I'm going to change it.